English translation of skins system article (#2)
elyby/docs: source/en/skins-system.rst

Skins system
------------

> On this page you'll find information about available endpoints of Ely.by's service of skins system. You can use any of

Reviewer: ```suggestion
On this page you'll find information about available endpoints of Ely.by's skins system service. You can use any of
```

them as an secondary or primary source of skins for your project.

> Ely.by's service of skins system provides `proxying of textures from Minecraft premium users <#textures-proxy>`_,

Reviewer: ```suggestion
Ely.by's skins system service provides `proxying of textures from Minecraft premium users <#textures-proxy>`_,
```

> which means that when using this service players will see both premium Minecraft users' skins and Ely.by users' skins.

Reviewer: ```suggestion
which means that using this service, your players will see both premium Minecraft users' skins and Ely.by users' skins.
```
Maybe you want to add "your players", if you are trying to market this service to servers holders.

We strive to comply with the official skins system and do not support ears and HD-skins. The system supports capes,

> but doesn't allow players to wear them on their own.

Reviewer: ```suggestion
but doesn't allow players to don them on their own.
```

> If you have suggestions for improvements of existing functionality, please

Reviewer: ```suggestion
If you have suggestions for improving the existing functionality, please
```

> `create a new Issue <https://github.com/elyby/chrly/issues/new>`_ in the

Reviewer: ```suggestion
`create a new Issue <https://github.com/elyby/chrly/issues/new>`_ at the
```

`Chrly project repository <https://github.com/elyby/chrly>`_.

Requests URLs
=============

.. note:: You can find more detailed information about the implementation of the skins system server in the
          `Chrly project repository <https://github.com/elyby/chrly>`_.

> The skins system is placed on the domain :samp:`http://skinsystem.ely.by`. The :samp:`nickname` parameter

Reviewer: ```suggestion
The skins system is located at the :samp:`http://skinsystem.ely.by` domain. The :samp:`nickname` parameter
```

> is case-insensitive.

Reviewer: "The :samp:`nickname` parameter is case-insensitive."

Feels a bit random/weird to learn about this in this paragraph.

The following endpoints are used to get information about textures:

.. _skin-request:
.. function:: /skins/{nickname}.png

>    This URL is responsible for skins downloading. The **nickname** param must be replaced with the player's nickname.

Reviewer: ```suggestion
   This URL is responsible for downloading skins. The **nickname** param must be replaced with the player's nickname.
```

   The :samp:`.png` extension can be omitted.

>    The server will return a :samp:`404` status response in case when textures aren't found.

Reviewer: ```suggestion
   If the texture wasn't found, the server would return a :samp:`404` status response.
```

.. _cape-request:
.. function:: /cloaks/{nickname}.png

>    This URL is responsible for capes downloading. The **nickname** param must be replaced with the player's nickname.

Reviewer: ```suggestion
   This URL is responsible for downloading capes. The **nickname** param must be replaced with the player's nickname.
```

   The :samp:`.png` extension can be omitted.

>    The server will return a :samp:`404` status response in case when textures aren't found.

Reviewer: ```suggestion
   If the texture wasn't found, the server would return a :samp:`404` status response.
```

.. function:: /textures/{nickname}

>    By this URL you can get textures for the **nickname** specified in the request. The result is a JSON document

Reviewer: ```suggestion
   Via this URL you can get textures for the **nickname** specified in the request. The result'll be a JSON document:
```

   of the following format:

   .. code-block:: javascript

      {
        "SKIN": {
          "url": "http://example.com/skin.png",
          "metadata": {
            "model": "slim"
          }
        },
        "CAPE": {
          "url": "http://example.com/cape.png"
        }
      }

>    Depending on available textures for the player, there may be no :samp:`SKIN` or :samp:`CAPE` fields.

Reviewer: ```suggestion
   Depending on the availability of textures for the player, fields :samp:`SKIN` or :samp:`CAPE` may be absent.
```

>    If the skin model isn't :samp:`slim`, the :samp:`metadata` field will also be omitted.

Reviewer: ```suggestion
   Unless the skin model is :samp:`slim`, the :samp:`metadata` field will be omitted.
```

>    The server will return an empty response with :samp:`204` status in case when textures aren't found.

Reviewer: ```suggestion
   The server will return an empty response with :samp:`204` status, if textures aren't found.
```
Maybe :samp:`404` status?

.. function:: /textures/signed/{nickname}

   This request is used in our `server skins system plugin <http://ely.by/server-skins-system>`_ to load textures with

>    the original Mojang's signature. The textures received in the reply can be transferred to the unmodified game client

Reviewer: ```suggestion
   the original Mojang's signature. The textures received this way can be transferred to the unmodified game client
```
Original was okay too, but I think this variant is more readable.

>    the original Mojang's signature. The textures received in the reply can be transferred to the unmodified game client

Reviewer: ```suggestion
   the original Mojang's signature. The textures received in the reply can be transferred to an unmodified game client
```

            "value": "eyJ0aW1lc3RhbXAiOjE0ODYzMzcyNTQ4NzIsInByb2ZpbGVJZCI6ImM0ZjFlNTZmNjFkMTQwYTc4YzMyOGQ5MTY2ZWVmOWU3IiwicHJvZmlsZU5hbWUiOiJXaHlZb3VSZWFkVGhpcyIsInRleHR1cmVzIjp7IlNLSU4iOnsidXJsIjoiaHR0cDovL3RleHR1cmVzLm1pbmVjcmFmdC5uZXQvdGV4dHVyZS83Mzk1NmE4ZTY0ZWU2ZDhlYzY1NmFkYmI0NDA0ZjhlYmZmMzQxMWIwY2I5MGIzMWNiNDc2ZWNiOTk2ZDNiOCJ9fX0="
          },
          {
            "name": "ely",
            "value": "but why are you asking?"
          }
        ]
      }

   By default textures proxying isn't used for this query. To enable it, add an additional GET parameter
   :samp:`?proxy=true`.

>    The server will return an empty response with :samp:`204` status in case when textures aren't found.

Reviewer: ```suggestion
   The server will return an empty response with :samp:`204` status, if textures aren't found.
```
Again, maybe 404 status? I have no idea.

Author: No, this is `204` status because all new Mojang's APIs use the same logic.

------------------------------------------------------------------------------------------------------------------------

> You can also pass a number of additional GET parameters when making any of the above requests. They will be used

Reviewer: ```suggestion
You can also pass a range of additional GET parameters while making any of the above requests. They will be used
```

to analyze the usage of the service by different versions of the game.

> :version: The version of the protocol that the skins are being requested. This is currently version :samp:`2`,

Reviewer: ```suggestion
:version: The version of the protocol by which skins will be requested. The current version is :samp:`2`,
```

>           i.e. you need to specify :samp:`version=2`.

Reviewer: ```suggestion
          so you are to specify :samp:`version=2`.
```

:minecraft_version: The version of Minecraft that the request is made from.

:authlib_version: The version of the Authlib used. This option is relevant for Minecraft versions 1.7.6+,

>                   where a separate library is used to load skins rather than an in-game code.

Reviewer: ```suggestion
                  where a separate library is used to load skins instead of in-game code.
```

> An example of a textures query with passing of parameters described above:

Reviewer: ```suggestion
Here is an example of a textures request with parameters described above:
```

.. code-block:: text

   http://skinsystem.ely.by/textures/erickskrauch?version=2&minecraft_version=1.14.0&authlib_version=1.5.25

Additional URLs
+++++++++++++++

You can also perform a skin and cape request by passing the nickname through the GET parameter. This feature is used

> to pass analytical parameters on the game versions up to 1.5.2, when the nickname is simply appended to the end of the

Reviewer: ```suggestion
to pass analytical parameters of game versions up to 1.5.2, where the nickname is simply appended to the end of the
```
Same is for the Russian version: "в которых" instead of "когда"

> line. To do this, the entire string is arranged in such a way that the last parameter was :samp:`name`, after appending

Reviewer: ```suggestion
line. To do this, the entire string is arranged in such a way that the last parameter is :samp:`name`, after appending
```

a nickname to which you get a full request string for textures.

.. function:: /skins?name={nickname}.png

   See the `skin request <#skin-request>`_.

.. function:: /cloaks?name={nickname}.png

   See the `cape request <#cape-request>`_.

> Example of requests for textures with parameters above:

Reviewer: ```suggestion
Examples of requests for textures with above parameters:
```

.. code-block:: text

   http://skinsystem.ely.by/skins?version=2&minecraft_version=1.5.2&name=erickskrauch.png
   http://skinsystem.ely.by/cloaks?version=2&minecraft_version=1.4.7&name=notch

.. _textures-proxy:

Textures proxying
=================

> Ely.by's service of skins system obtains textures from the official skin system in case no information about textures

Reviewer: ```suggestion
Ely.by's skins system service obtains textures from the official skin system in a case where no information about textures
```

> for the requested user name was found in the database. The request will also be proxied if a skin entry is found,

Reviewer: ```suggestion
for the requested username was found in the database. The request will also be proxied if a skin entry is found,
```

but it's default.

> To improve the throughput of the proxying algorithm, information about textures is cached in the 2 stages:

Reviewer: ```suggestion
To improve the throughput of the proxying algorithm, information about textures is cached in 2 stages:
```

* Player's names and UUIDs matches are stored
  `for 30 days <https://help.mojang.com/customer/portal/articles/928638#targetText=How%20often%20can%20I%20change%20my%20username%3F>`_.

> * Information about textures is updated not more often than

Reviewer: ```suggestion
* Information about textures isn't updated more often than
```

  `once a minute <https://wiki.vg/Mojang_API#UUID_-.3E_Profile_.2B_Skin.2FCape>`_.

> If you own a Minecraft premium account, but your nickname is busy, please contact our

Reviewer: I think this information should also be doubled in an article for players, if it isn't already.

Author: It's already done: https://account.ely.by/rules#rule-2-5

`support team <http://ely.by/site/contact>`_ and after a short check we'll pass the nickname on to you.

Ready-made implementations
==========================

> Ready-made patch implementations and installation instructions can be found in the

Reviewer: ```suggestion
Ready-made patch implementations and installation instructions can be found at the
```

`create a new Issue <https://github.com/elyby/chrly/issues/new>`_ at the
`Chrly project repository <https://github.com/elyby/chrly>`_.

.. note:: You can find more detailed information about the implementation of the skins system server in the
          `Chrly project repository <https://github.com/elyby/chrly>`_.

Requests URLs
=============

The skins system is located at the :samp:`http://skinsystem.ely.by` domain.

In all queries, the :samp:`nickname` param must be replaced by the player's name. The value is case-insensitive.

.. _skin-request:
.. function:: /skins/{nickname}.png

>    URL for downloading skin texture. The :samp:`.png` extension can be omitted. If textures aren't found,

Reviewer: "**a** skin texture"

   the server will return a :samp:`404` status response.

.. _cape-request:
.. function:: /cloaks/{nickname}.png

>    URL for downloading cape texture. The :samp:`.png` extension can be omitted. If textures aren't found,

Reviewer: "**a** cape texture"
